client/my-sites/plans-grid/hooks/npm-ready/data-store/use-grid-plans.tsx

```diff
@@ -25,6 +25,7 @@ import {
 	isEcommercePlan,
 	TYPE_P2_PLUS,
 } from '@automattic/calypso-products';
+import useAddOnCheckoutLink from 'calypso/my-sites/add-ons/hooks/use-add-on-checkout-link';
```

Comment:
This is a local dependency we are introducing here. Let's either move it to a respective package and import from there or surface it differently into this hook :-)

Reply:
p.s. looking at the code for `useAddOnCheckoutLink`, it should be easy to just copy over into a respective package by passing siteId in the props. 

If that's the direction, would that be `packages/data-stores/add-ons`, or is it `packages/add-ons`?

Reply:
Ah thanks for catching this @chriskmnds! My mistake 🙏

Reply:
OK. Please read my other comments too. I don't think we should be passing this through as a callback. It complicates things further, and there seems to be a cleaner approach to putting these together.

From the looks of it, it sounds like we can instead compile the storage add-ons _with_ a checkout URL from the existing `useStorageAddOns` hook. In which case, it becomes less of an immediate concern to migrate or refactor the `useAddOnCheckoutLink` hook. 

Hope this makes sense?

Reply:
>  OK. Please read my other comments too.

Yeah I'm seeing the other comments now, and they make sense. 🙂

I'll hold off on responding individually for the time being until I've read everything as a whole

Reply:
> From the looks of it, it sounds like we can instead compile the storage add-ons with a checkout URL from the existing useStorageAddOns hook. In which case, it becomes less of an immediate concern to migrate or refactor the useAddOnCheckoutLink hook.

> IF it makes sense. Otherwise, I think moving useAddOnCheckoutLink to the respective add-ons package might be better than passing in a callback here 🤔

I'll probably move forward with both ideas 👍

I like the idea of generating storage add-ons with their respective checkout links for ease of use, especially because the requisite change is simple. Beyond that though, moving `useAddOnCheckoutLink` also looks like an easy task and gets us one step closer to having add-ons in a self-contained package. It's as good a reason as any to kick off the our migration, so I'll opt to handle that in an immediate follow-up

Reply:
> From the looks of it, it sounds like we can instead compile the storage add-ons with a checkout URL from the existing useStorageAddOns hook. In which case, it becomes less of an immediate concern to migrate or refactor the useAddOnCheckoutLink hook.

https://github.com/Automattic/wp-calypso/pull/83005/commits/5ca53a6f37850ac4389026ec1fa9d482f8d7d22b